mlkem/ntt.c

```diff
@@ -57,9 +58,20 @@ const int16_t zetas[128] = {
  * Name:        poly_ntt
  *
  * Description: Computes negacyclic number-theoretic transform (NTT) of
- *              a polynomial in place;
- *              inputs assumed to be in normal order, output in bitreversed
- *order
+ *              a polynomial in place.
+ *
+ *              The input is assumed to be in normal order and
+ *              coefficient-wise bound by KYBER_Q in absolute value.
+ *
+ *              The output polynomail is in bitreversed order, and
+ *              coefficient-wise bound by NTT_BOUND in absolute value.
+ *
+ *              Values of NTT_BOUND:
+ *              - C: 16546
```

Comment:
I'm not sure I like this, since it complicates the C/Asm boundary with potentially different contracts for each back-end. Can't we reduce here so that _all_ backends always return reduced coefficients that are always in the same range?

Reply:
Yes, I want a _single_ "worst-case" contract between C and ASM that _all_ implementations satisfy. But the way it's currently phrased, that doesn't come across. I'll try again.

Reply:
Done